git-rci.ps1

```diff
@@ -0,0 +1,138 @@
+
+        for ($i = 0; $i -lt $filteredBranches.Count; $i++) {
+            $isSelected = $false
+            if ($filteredBranches[$i] -in $selectedBranches) { $isSelected = $true }
+
+            if ($i -eq $currentIndex) {
+                Write-Host "->" -NoNewline -ForegroundColor Yellow
+            }
+            else {
+                Write-Host "  " -NoNewline
+            }
+
+            Write-Host ("[" + ($isSelected ? "x" : " ") + "]") -NoNewline -ForegroundColor Yellow
+            Write-Host (" $($filteredBranches[$i].branch)") -ForegroundColor Green
+        }
+
+        Write-Host ""
+        Write-Host "Selected Branches:" -ForegroundColor Cyan
+        foreach ($branch in $selectedBranches) {
+            Write-Host "  $($branch.branch)"
+        }
+
+        Write-Host ""
+        Write-Host "Filter: '$filterText'" -ForegroundColor Cyan
+
+        $key = $host.UI.RawUI.ReadKey("NoEcho,IncludeKeyDown")
+
+        switch ($key.VirtualKeyCode) {
+            38 { if ($currentIndex -gt 0) { $currentIndex-- } }         # Up arrow
+            40 { if ($currentIndex -lt $filteredBranches.Count - 1) { $currentIndex++ } } # Down arrow
+            13 {
+                $selectedBranch = $filteredBranches[$currentIndex]
+                if ($selectedBranches.Contains($selectedBranch)) {
+                    $selectedBranches.Remove($selectedBranch)
+                }
+                else {
+                    $selectedBranches.Add($selectedBranch)
+                }
+            }
+            27 { 
+                return $selectedBranches 
+            }
+            8 {  # Backspace
+                if ($filterText.Length -gt 0) {
+                    $filterText = $filterText.Substring(0, $filterText.Length - 1)
+                }
+            }
+            default {
+                if ($key.Character -match "\w") {
+                    $filterText += $key.Character
+                }
+            }
+        }
+    }
+}
+
+
+. $PSScriptRoot/config/core/split-string.ps1
+. $PSScriptRoot/config/core/coalesce.ps1
+. $PSScriptRoot/config/branch-utils/ConvertTo-BranchName.ps1
+. $PSScriptRoot/config/git/Get-Configuration.ps1
+. $PSScriptRoot/config/git/Update-Git.ps1
+. $PSScriptRoot/config/git/Assert-CleanWorkingDirectory.ps1
+. $PSScriptRoot/config/git/Select-Branches.ps1
+. $PSScriptRoot/config/git/Invoke-PreserveBranch.ps1
+. $PSScriptRoot/config/git/Invoke-CreateBranch.ps1
+. $PSScriptRoot/config/git/Invoke-CheckoutBranch.ps1
+Import-Module "$PSScriptRoot/config/git/Invoke-MergeBranches.psm1";
+. $PSScriptRoot/config/git/Set-UpstreamBranches.ps1
+
+$config = Get-Configuration
+
+if (-not $noFetch) {
+    Update-Git -config $config
+}
+
+$selectedBranches = @()
+
+Assert-CleanWorkingDirectory
+$allBranches = Select-Branches -config $config
+$allBranches = $allBranches | Where-Object { $_.branch -ne 'main' -and $_.branch -ne '_upstream' }
```

Comment:
This should use the `$config` object instead of hard-coding these branches:

    $_.branch -ne $config.defaultServiceLine -and $_.branch -ne $config.upstreamBranch